src/main/java/com/ryanharter/auto/value/moshi/AutoValueMoshiAdapterFactoryProcessor.java

```diff
@@ -66,22 +78,42 @@ public boolean process(Set<? extends TypeElement> annotations, RoundEnvironment
     }
 
     if (!elements.isEmpty()) {
-      TypeSpec jsonAdapterFactory = createJsonAdapterFactory(elements);
-      JavaFile file = JavaFile.builder("com.ryanharter.auto.value.moshi", jsonAdapterFactory).build();
-      try {
-        file.writeTo(processingEnv.getFiler());
-      } catch (IOException e) {
-        processingEnv.getMessager().printMessage(Diagnostic.Kind.ERROR, "Failed to write TypeAdapterFactory: " + e.getLocalizedMessage());
+      Set<? extends Element> adaptorFactories = roundEnv.getElementsAnnotatedWith(MoshiAdapterFactory.class);
+      if (adaptorFactories.isEmpty()) {
+        error(elements.get(0), "Missing MoshiAdapterFactory class. In order to generate the class, you need to create an abstract class with the @MoshiAdapterFactory annotation that implements JsonAdapter.Factory.");
```

Comment:
If I'm reading this right, this will fail to compile if you don't have a `@MoshiAdapterFactory` annotated class, but that seems a bit harsh.  I would expect the adapter factory should be optional, not required, since the user can still access the `TypeAdapter`s directly from the AutoValue classes, without using a single, global adapter factory.